pip/download.py

```diff
@@ -198,6 +203,31 @@ def parse_credentials(self, netloc):
             return userinfo, None
         return None, None
 
+    @classmethod
```

Comment:
Initially I intended to make it a class variable but had issues with the implementation and changed it a class method. 
It can work as a function too; that is, no `@property`. is the advantage to avoid the extra pair of parenthesis `self.authlib()(....)` when it s being used?